xscen/utils.py

```diff
@@ -106,13 +116,39 @@ def stack_drop_nans(
     --------
     unstack_fill_nan : The inverse operation.
     """
+
+    original_shape = f"{len(ds.lon)}x{len(ds.lat)}"
```

Comment:
Je sais pas à quel point c'était utile, mais l'ancienne version de cette fonction était indépendante des noms de dimensions. On utilise `mask.dims`.
Est-ce que ça serait à garder?
Si oui, je propose:
```suggestion
    original_shape = "x".join(map(str, mask.shape))
```
et il faudrait rendre le reste du code générique aux dims de `mask`...

Et si non, il faudrait juste le dire dans la fonction qu'on s'attend à `lat` et `lon`.

Reply:
ah ben oui ça serait bien de garder cette flexibilité là. Je vais faire les changements.

Reply:
The changes I have added will fail if one of the dimensions of ds_stack have an unrelated attrs also named `original_shape`, but that feels very unlikely...